api/src/main/java/org/openmrs/module/ModuleFactory.java

```diff
@@ -171,9 +171,10 @@ public static void loadModules() {
 		if (log.isDebugEnabled()) {
 			log.debug("Loading modules from: " + modulesFolder.getAbsolutePath());
 		}
-		
-		if (modulesFolder.isDirectory()) {
-			loadModules(Arrays.asList(Objects.requireNonNull(modulesFolder.listFiles())));
+
+		File[] fileList = modulesFolder.listFiles();
+		if (modulesFolder.isDirectory() && fileList != null) {
+			loadModules(Arrays.asList(fileList));
 		} else {
 			log.error("modules folder: '" + modulesFolder.getAbsolutePath() + "' is not a valid directory");
```

Comment:
now the log message also includes a directory which are empty which I assume is still a valid directory.

Reply:
moduleFolder.listFiles() returns null only when directory is not valid